src/Raven.Client/Documents/Subscriptions/SubscriptionCriteria.cs

```diff
@@ -113,8 +113,36 @@ public override void ConvertToJavascript(JavascriptConversionContext context)
                     }
 
                     return;
-                }                
-                
+                }
+
+                var newExp = context.Node as NewExpression;
+
+                if (newExp != null && newExp.Type == typeof(DateTime))
+                {
+                    var args = newExp.Arguments;
+                    var argsStr = string.Empty;
```

Comment:
Can you build it inline? If we call it each time, that can cause a lot of allocations